README.md

```diff
@@ -25,7 +25,7 @@ Features
 Build the javascript version
 ----------------------------
 
-You need to make sure you have both [emscripten](https://emscripten.org/docs/getting_started/downloads.html) and [sconstruct](https://pypi.org/project/SCons/) installed.
+You need to make sure you have both [emscripten](https://emscripten.org/docs/getting_started/downloads.html) 1.40.1 (it does not work with latest version) and [sconstruct](https://pypi.org/project/SCons/) installed.
```

Comment:
Actually it works at least on 2.0.11

Reply:
Yes, I tested ok with 2.0.11, the current latest version is 3.x ([3.1.51](https://github.com/emscripten-core/emsdk/releases/tag/3.1.51)). and I also tried the last 2.x ([2.0.34](https://github.com/emscripten-core/emsdk/tags?after=3.1.0)), it failed with the following errors.

<details><summary>Details</summary>

```
@liudonghua123 ➜ /workspaces/stellarium-web-engine (master) $ make clean
scons -c
scons: Reading SConscript files ...
scons: done reading SConscript files.
scons: Cleaning targets ...
Removed build/ext_src/webp/src/dec/alpha_dec.o
Removed build/ext_src/webp/src/dec/buffer_dec.o
Removed build/ext_src/webp/src/dec/frame_dec.o
Removed build/ext_src/webp/src/dec/idec_dec.o
Removed build/ext_src/webp/src/dec/io_dec.o
Removed build/ext_src/webp/src/dec/quant_dec.o
Removed build/ext_src/webp/src/dec/tree_dec.o
Removed build/ext_src/webp/src/dec/vp8_dec.o
Removed build/ext_src/webp/src/dec/vp8l_dec.o
Removed build/ext_src/webp/src/dec/webp_dec.o
Removed build/ext_src/webp/src/dsp/alpha_processing.o
Removed build/ext_src/webp/src/dsp/cpu.o
Removed build/ext_src/webp/src/dsp/dec.o
Removed build/ext_src/webp/src/dsp/dec_clip_tables.o
Removed build/ext_src/webp/src/dsp/filters.o
Removed build/ext_src/webp/src/dsp/lossless.o
Removed build/ext_src/webp/src/dsp/rescaler.o
Removed build/ext_src/webp/src/dsp/upsampling.o
Removed build/ext_src/webp/src/dsp/yuv.o
Removed build/ext_src/webp/src/utils/bit_reader_utils.o
Removed build/ext_src/webp/src/utils/color_cache_utils.o
Removed build/ext_src/webp/src/utils/filters_utils.o
Removed build/ext_src/webp/src/utils/huffman_utils.o
Removed build/ext_src/webp/src/utils/quant_levels_dec_utils.o
Removed build/ext_src/webp/src/utils/random_utils.o
Removed build/ext_src/webp/src/utils/rescaler_utils.o
Removed build/ext_src/webp/src/utils/thread_utils.o
Removed build/ext_src/webp/src/utils/utils.o
Removed build/ext_src/zlib/adler32.o
Removed build/ext_src/zlib/compress.o
Removed build/ext_src/zlib/crc32.o
Removed build/ext_src/zlib/deflate.o
Removed build/ext_src/zlib/gzclose.o
Removed build/ext_src/zlib/gzlib.o
Removed build/ext_src/zlib/gzread.o
Removed build/ext_src/zlib/gzwrite.o
Removed build/ext_src/zlib/infback.o
Removed build/ext_src/zlib/inffast.o
Removed build/ext_src/zlib/inflate.o
Removed build/ext_src/zlib/inftrees.o
Removed build/ext_src/zlib/trees.o
Removed build/ext_src/zlib/uncompr.o
Removed build/ext_src/zlib/zutil.o
Removed build/src/algos/bv_to_rgb.o
Removed build/src/algos/cst-boundaries.o
Removed build/src/algos/deltat.o
Removed build/src/algos/format.o
Removed build/src/algos/gust86.o
Removed build/src/algos/healpix.o
Removed build/src/algos/l1.o
Removed build/src/algos/moon.o
Removed build/src/algos/orbit.o
Removed build/src/algos/pluto.o
Removed build/src/algos/refraction.o
Removed build/src/algos/satrings.o
Removed build/src/algos/tass17.o
Removed build/src/algos/utctt.o
Removed build/src/areas.o
Removed build/src/args.o
Removed build/src/assets.o
Removed build/src/core.o
Removed build/src/designation.o
Removed build/src/eph-file.o
Removed build/src/events.o
Removed build/src/frames.o
Removed build/src/geojson_parser.o
Removed build/src/hip.o
Removed build/src/hips.o
Removed build/src/layer.o
Removed build/src/libtess2.o
Removed build/src/line_mesh.o
Removed build/src/module.o
Removed build/src/modules/atmosphere.o
Removed build/src/modules/cardinal.o
Removed build/src/modules/circle.o
Removed build/src/modules/comets.o
Removed build/src/modules/constellations.o
Removed build/src/modules/coordinates.o
Removed build/src/modules/debug.o
Removed build/src/modules/drag_selection.o
Removed build/src/modules/dso.o
Removed build/src/modules/dss.o
Removed build/src/modules/geojson.o
Removed build/src/modules/labels.o
Removed build/src/modules/landscape.o
Removed build/src/modules/lines.o
Removed build/src/modules/meteors.o
Removed build/src/modules/milkyway.o
Removed build/src/modules/minorplanets.o
Removed build/src/modules/movements.o
Removed build/src/modules/photos.o
Removed build/src/modules/planets.o
Removed build/src/modules/pointer.o
Removed build/src/modules/satellites.o
Removed build/src/modules/skycultures.o
Removed build/src/modules/stars.o
Removed build/src/mpc.o
Removed build/src/navigation.o
Removed build/src/obj.o
Removed build/src/observer.o
Removed build/src/otypes.o
Removed build/src/painter.o
Removed build/src/projection.o
Removed build/src/projections/proj_hammer.o
Removed build/src/projections/proj_mercator.o
Removed build/src/projections/proj_mollweide.o
Removed build/src/projections/proj_perspective.o
Removed build/src/projections/proj_stereographic.o
Removed build/src/render_gl.o
Removed build/src/sgp4.o
Removed build/src/shader_cache.o
Removed build/src/skybrightness.o
Removed build/src/skyculture.o
Removed build/src/sprintf.o
Removed build/src/swe.o
Removed build/src/symbols.o
Removed build/src/system.o
Removed build/src/telescope.o
Removed build/src/tests.o
Removed build/src/tonemapper.o
Removed build/src/utils/cache.o
Removed build/src/utils/fader.o
Removed build/src/utils/fps.o
Removed build/src/utils/gesture.o
Removed build/src/utils/gl.o
Removed build/src/utils/mesh.o
Removed build/src/utils/progressbar.o
Removed build/src/utils/request.o
Removed build/src/utils/request_js.o
Removed build/src/utils/swe_utils.o
Removed build/src/utils/texture.o
Removed build/src/utils/utf8.o
Removed build/src/utils/utils_json.o
Removed build/src/utils/vec.o
Removed build/src/utils/worker.o
Removed build/src/uv_map.o
Removed build/stellarium-web-engine.js
scons: done cleaning targets.
@liudonghua123 ➜ /workspaces/stellarium-web-engine (master) $ emsdk/emsdk list

The *recommended* precompiled SDK download is 3.1.51 (4f416d92fbff66ce79901cfc8263768f1b25dd3e).

To install/activate it use:
         latest

This is equivalent to installing/activating:
         3.1.51             

All recent (non-legacy) installable versions are:
         3.1.51    
         3.1.51-asserts    
         3.1.50    
         3.1.50-asserts    
         3.1.49    
         3.1.49-asserts    
         3.1.48    
         3.1.48-asserts    
         3.1.47    
         3.1.47-asserts    
         3.1.46    
         3.1.46-asserts    
         3.1.45    
         3.1.45-asserts    
         3.1.44    
         3.1.44-asserts    
         3.1.43    
         3.1.43-asserts    
         3.1.42    
         3.1.42-asserts    
         3.1.41    
         3.1.41-asserts    
         3.1.40    
         3.1.40-asserts    
         3.1.39    
         3.1.39-asserts    
         3.1.38    
         3.1.38-asserts    
         3.1.37    
         3.1.37-asserts    
         3.1.36    
         3.1.36-asserts    
         3.1.35    
         3.1.35-asserts    
         3.1.34    
         3.1.34-asserts    
         3.1.33    
         3.1.33-asserts    
         3.1.32    
         3.1.32-asserts    
         3.1.31    
         3.1.31-asserts    
         3.1.30    
         3.1.30-asserts    
         3.1.29    
         3.1.29-asserts    
         3.1.28    
         3.1.28-asserts    
         3.1.27    
         3.1.27-asserts    
         3.1.26    
         3.1.26-asserts    
         3.1.25    
         3.1.25-asserts    
         3.1.24    
         3.1.24-asserts    
         3.1.23    
         3.1.23-asserts    
         3.1.22    
         3.1.22-asserts    
         3.1.21    
         3.1.21-asserts    
         3.1.20    
         3.1.20-asserts    
         3.1.19    
         3.1.19-asserts    
         3.1.18    
         3.1.18-asserts    
         3.1.17    
         3.1.17-asserts    
         3.1.16    
         3.1.16-asserts    
         3.1.15    
         3.1.15-asserts    
         3.1.14    
         3.1.14-asserts    
         3.1.13    
         3.1.13-asserts    
         3.1.12    
         3.1.12-asserts    
         3.1.11    
         3.1.11-asserts    
         3.1.10    
         3.1.10-asserts    
         3.1.9    
         3.1.9-asserts    
         3.1.8    
         3.1.8-asserts    
         3.1.7    
         3.1.7-asserts    
         3.1.6    
         3.1.6-asserts    
         3.1.5    
         3.1.5-asserts    
         3.1.4    
         3.1.4-asserts    
         3.1.3    
         3.1.3-asserts    
         3.1.2    
         3.1.2-asserts    
         3.1.1    
         3.1.1-asserts    
         3.1.0    
         3.1.0-asserts    
         3.0.1    
         3.0.1-asserts    
         3.0.0    
         3.0.0-asserts    
         2.0.34    INSTALLED
         2.0.33    
         2.0.32    
         2.0.31    
         2.0.31-asserts    
         2.0.30    
         2.0.30-asserts    
         2.0.29    
         2.0.29-lto    
         2.0.28    
         2.0.28-lto    
         2.0.27    
         2.0.27-lto    
         2.0.26    
         2.0.26-lto    
         2.0.25    
         2.0.24    
         2.0.23    
         2.0.23-lto    
         2.0.22    
         2.0.21    
         2.0.20    
         2.0.20-lto    
         2.0.19    
         2.0.19-lto    
         2.0.18    
         2.0.17    
         2.0.16    
         2.0.15    
         2.0.14    
         2.0.13    
         2.0.12    
         2.0.11    
         2.0.10    
         2.0.9    
         2.0.8    
         2.0.7    
         2.0.6    
         2.0.5    
         2.0.4    
         2.0.3    
         2.0.2    
         2.0.1    
         2.0.0    
         1.40.1    
         1.40.0    
         1.39.20    
         1.39.19    
         1.39.18    
         1.39.17    
         1.39.16    
         1.39.15    
         1.39.14    
         1.39.13    
         1.39.12    
         1.39.11    
         1.39.10    
         1.39.9    
         1.39.8    
         1.39.7    
         1.39.6    
         1.39.5    
         1.39.4    
         1.39.3    
         1.39.2    
         1.39.1    
         1.39.0    
         1.38.48    
         1.38.47    
         1.38.46    
         1.38.45    
         1.38.44    
         1.38.43    
         1.38.42    
         1.38.41    
         1.38.40    
         1.38.39    
         1.38.38    
         1.38.37    
         1.38.36    
         1.38.35    
         1.38.34    
         1.38.33    

The additional following precompiled SDKs are also available for download:
         sdk-releases-4f416d92fbff66ce79901cfc8263768f1b25dd3e-64bit
         sdk-releases-9035c99beb760aa0ea381bdf11abf440d88bb451-64bit

The following SDKs can be compiled from source:
         sdk-main-64bit           
         sdk-main-32bit           

The following precompiled tool packages are available for download:
           releases-4f416d92fbff66ce79901cfc8263768f1b25dd3e-64bit
           releases-9035c99beb760aa0ea381bdf11abf440d88bb451-64bit
           node-14.18.2-64bit       
           node-14.15.5-64bit       
           node-15.14.0-64bit       
    (*)    node-16.20.0-64bit           INSTALLED
           emscripten-tag-1.38.30-32bit
           emscripten-tag-1.38.31-32bit
           emscripten-tag-1.38.30-64bit
           emscripten-tag-1.38.31-64bit
           emscripten-1.38.30       
           emscripten-1.38.31       

The following tools can be compiled from source:
           llvm-git-main-32bit      
           llvm-git-main-64bit      
           binaryen-tag-1.38.30-32bit
           binaryen-tag-1.38.31-32bit
           binaryen-tag-1.38.30-64bit
           binaryen-tag-1.38.31-64bit
           emscripten-main-32bit    
           emscripten-main-64bit    
           binaryen-main-32bit      
           binaryen-main-64bit      
           ninja-git-release-64bit  
           ccache-git-emscripten-64bit

Items marked with * are activated for the current user.
Items marked with (*) are selected for use, but your current shell environment is not configured to use them. Type "source ./emsdk_env.sh" to set up your current shell to use them.

To access the historical archived versions, type 'emsdk list --old'

Run "git pull" to pull in the latest list.
@liudonghua123 ➜ /workspaces/stellarium-web-engine (master) $ make js
emscons scons -j8 mode=release
scons: Reading SConscript files ...
scons: done reading SConscript files.
scons: Building targets ...
/workspaces/stellarium-web-engine/emsdk/upstream/emscripten/tools/scons/site_scons/site_tools/emscripten/../../../../../emcc -o build/ext_src/erfa/erfa.o -c -Wall -std=gnu11 -Wno-unknown-pragmas -D_GNU_SOURCE -Wno-missing-braces -DHAVE_UNISTD_H -DMD4C_USE_UTF8 -Werror -DNDEBUG -include config.h -Wno-initializer-overrides -DNO_LIBCURL -DNO_ARGP "-DGLES2 1" -s MODULARIZE=1 -s EXPORT_NAME=StelWebEngine -s ALLOW_MEMORY_GROWTH=1 -s ALLOW_TABLE_GROWTH=1 --pre-js src/js/pre.js --pre-js src/js/obj.js --pre-js src/js/geojson.js --pre-js src/js/canvas.js -s RESERVED_FUNCTION_POINTERS=10 -O3 -s USE_WEBGL2=1 -s NO_EXIT_RUNTIME=1 -s "EXPORTED_FUNCTIONS=[]" -s "EXTRA_EXPORTED_RUNTIME_METHODS=['ALLOC_NORMAL','GL','UTF8ToString','_free','_malloc','addFunction','allocate','ccall','cwrap','getValue','intArrayFromString','lengthBytesUTF8','removeFunction','setValue','stringToUTF8','writeAsciiToMemory','writeArrayToMemory']" -s FILESYSTEM=0 -Isrc -Iext_src/erfa -Iext_src/json -Iext_src/uthash -Iext_src/stb -Iext_src/zlib -Iext_src/inih -Iext_src/nanovg -Iext_src/md4c -Iext_src/webp -Iext_src/webp/src ext_src/erfa/erfa.c
/workspaces/stellarium-web-engine/emsdk/upstream/emscripten/tools/scons/site_scons/site_tools/emscripten/../../../../../emcc -o build/ext_src/inih/ini.o -c -Wall -std=gnu11 -Wno-unknown-pragmas -D_GNU_SOURCE -Wno-missing-braces -DHAVE_UNISTD_H -DMD4C_USE_UTF8 -Werror -DNDEBUG -include config.h -Wno-initializer-overrides -DNO_LIBCURL -DNO_ARGP "-DGLES2 1" -s MODULARIZE=1 -s EXPORT_NAME=StelWebEngine -s ALLOW_MEMORY_GROWTH=1 -s ALLOW_TABLE_GROWTH=1 --pre-js src/js/pre.js --pre-js src/js/obj.js --pre-js src/js/geojson.js --pre-js src/js/canvas.js -s RESERVED_FUNCTION_POINTERS=10 -O3 -s USE_WEBGL2=1 -s NO_EXIT_RUNTIME=1 -s "EXPORTED_FUNCTIONS=[]" -s "EXTRA_EXPORTED_RUNTIME_METHODS=['ALLOC_NORMAL','GL','UTF8ToString','_free','_malloc','addFunction','allocate','ccall','cwrap','getValue','intArrayFromString','lengthBytesUTF8','removeFunction','setValue','stringToUTF8','writeAsciiToMemory','writeArrayToMemory']" -s FILESYSTEM=0 -Isrc -Iext_src/erfa -Iext_src/json -Iext_src/uthash -Iext_src/stb -Iext_src/zlib -Iext_src/inih -Iext_src/nanovg -Iext_src/md4c -Iext_src/webp -Iext_src/webp/src ext_src/inih/ini.c
/workspaces/stellarium-web-engine/emsdk/upstream/emscripten/tools/scons/site_scons/site_tools/emscripten/../../../../../emcc -o build/ext_src/json/json-builder.o -c -Wall -std=gnu11 -Wno-unknown-pragmas -D_GNU_SOURCE -Wno-missing-braces -DHAVE_UNISTD_H -DMD4C_USE_UTF8 -Werror -DNDEBUG -include config.h -Wno-initializer-overrides -DNO_LIBCURL -DNO_ARGP "-DGLES2 1" -s MODULARIZE=1 -s EXPORT_NAME=StelWebEngine -s ALLOW_MEMORY_GROWTH=1 -s ALLOW_TABLE_GROWTH=1 --pre-js src/js/pre.js --pre-js src/js/obj.js --pre-js src/js/geojson.js --pre-js src/js/canvas.js -s RESERVED_FUNCTION_POINTERS=10 -O3 -s USE_WEBGL2=1 -s NO_EXIT_RUNTIME=1 -s "EXPORTED_FUNCTIONS=[]" -s "EXTRA_EXPORTED_RUNTIME_METHODS=['ALLOC_NORMAL','GL','UTF8ToString','_free','_malloc','addFunction','allocate','ccall','cwrap','getValue','intArrayFromString','lengthBytesUTF8','removeFunction','setValue','stringToUTF8','writeAsciiToMemory','writeArrayToMemory']" -s FILESYSTEM=0 -Isrc -Iext_src/erfa -Iext_src/json -Iext_src/uthash -Iext_src/stb -Iext_src/zlib -Iext_src/inih -Iext_src/nanovg -Iext_src/md4c -Iext_src/webp -Iext_src/webp/src ext_src/json/json-builder.c
/workspaces/stellarium-web-engine/emsdk/upstream/emscripten/tools/scons/site_scons/site_tools/emscripten/../../../../../emcc -o build/ext_src/json/json.o -c -Wall -std=gnu11 -Wno-unknown-pragmas -D_GNU_SOURCE -Wno-missing-braces -DHAVE_UNISTD_H -DMD4C_USE_UTF8 -Werror -DNDEBUG -include config.h -Wno-initializer-overrides -DNO_LIBCURL -DNO_ARGP "-DGLES2 1" -s MODULARIZE=1 -s EXPORT_NAME=StelWebEngine -s ALLOW_MEMORY_GROWTH=1 -s ALLOW_TABLE_GROWTH=1 --pre-js src/js/pre.js --pre-js src/js/obj.js --pre-js src/js/geojson.js --pre-js src/js/canvas.js -s RESERVED_FUNCTION_POINTERS=10 -O3 -s USE_WEBGL2=1 -s NO_EXIT_RUNTIME=1 -s "EXPORTED_FUNCTIONS=[]" -s "EXTRA_EXPORTED_RUNTIME_METHODS=['ALLOC_NORMAL','GL','UTF8ToString','_free','_malloc','addFunction','allocate','ccall','cwrap','getValue','intArrayFromString','lengthBytesUTF8','removeFunction','setValue','stringToUTF8','writeAsciiToMemory','writeArrayToMemory']" -s FILESYSTEM=0 -Isrc -Iext_src/erfa -Iext_src/json -Iext_src/uthash -Iext_src/stb -Iext_src/zlib -Iext_src/inih -Iext_src/nanovg -Iext_src/md4c -Iext_src/webp -Iext_src/webp/src ext_src/json/json.c
/workspaces/stellarium-web-engine/emsdk/upstream/emscripten/tools/scons/site_scons/site_tools/emscripten/../../../../../emcc -o build/ext_src/md4c/entity.o -c -Wall -std=gnu11 -Wno-unknown-pragmas -D_GNU_SOURCE -Wno-missing-braces -DHAVE_UNISTD_H -DMD4C_USE_UTF8 -Werror -DNDEBUG -include config.h -Wno-initializer-overrides -DNO_LIBCURL -DNO_ARGP "-DGLES2 1" -s MODULARIZE=1 -s EXPORT_NAME=StelWebEngine -s ALLOW_MEMORY_GROWTH=1 -s ALLOW_TABLE_GROWTH=1 --pre-js src/js/pre.js --pre-js src/js/obj.js --pre-js src/js/geojson.js --pre-js src/js/canvas.js -s RESERVED_FUNCTION_POINTERS=10 -O3 -s USE_WEBGL2=1 -s NO_EXIT_RUNTIME=1 -s "EXPORTED_FUNCTIONS=[]" -s "EXTRA_EXPORTED_RUNTIME_METHODS=['ALLOC_NORMAL','GL','UTF8ToString','_free','_malloc','addFunction','allocate','ccall','cwrap','getValue','intArrayFromString','lengthBytesUTF8','removeFunction','setValue','stringToUTF8','writeAsciiToMemory','writeArrayToMemory']" -s FILESYSTEM=0 -Isrc -Iext_src/erfa -Iext_src/json -Iext_src/uthash -Iext_src/stb -Iext_src/zlib -Iext_src/inih -Iext_src/nanovg -Iext_src/md4c -Iext_src/webp -Iext_src/webp/src ext_src/md4c/entity.c
/workspaces/stellarium-web-engine/emsdk/upstream/emscripten/tools/scons/site_scons/site_tools/emscripten/../../../../../emcc -o build/ext_src/md4c/md4c-html.o -c -Wall -std=gnu11 -Wno-unknown-pragmas -D_GNU_SOURCE -Wno-missing-braces -DHAVE_UNISTD_H -DMD4C_USE_UTF8 -Werror -DNDEBUG -include config.h -Wno-initializer-overrides -DNO_LIBCURL -DNO_ARGP "-DGLES2 1" -s MODULARIZE=1 -s EXPORT_NAME=StelWebEngine -s ALLOW_MEMORY_GROWTH=1 -s ALLOW_TABLE_GROWTH=1 --pre-js src/js/pre.js --pre-js src/js/obj.js --pre-js src/js/geojson.js --pre-js src/js/canvas.js -s RESERVED_FUNCTION_POINTERS=10 -O3 -s USE_WEBGL2=1 -s NO_EXIT_RUNTIME=1 -s "EXPORTED_FUNCTIONS=[]" -s "EXTRA_EXPORTED_RUNTIME_METHODS=['ALLOC_NORMAL','GL','UTF8ToString','_free','_malloc','addFunction','allocate','ccall','cwrap','getValue','intArrayFromString','lengthBytesUTF8','removeFunction','setValue','stringToUTF8','writeAsciiToMemory','writeArrayToMemory']" -s FILESYSTEM=0 -Isrc -Iext_src/erfa -Iext_src/json -Iext_src/uthash -Iext_src/stb -Iext_src/zlib -Iext_src/inih -Iext_src/nanovg -Iext_src/md4c -Iext_src/webp -Iext_src/webp/src ext_src/md4c/md4c-html.c
/workspaces/stellarium-web-engine/emsdk/upstream/emscripten/tools/scons/site_scons/site_tools/emscripten/../../../../../emcc -o build/ext_src/md4c/md4c.o -c -Wall -std=gnu11 -Wno-unknown-pragmas -D_GNU_SOURCE -Wno-missing-braces -DHAVE_UNISTD_H -DMD4C_USE_UTF8 -Werror -DNDEBUG -include config.h -Wno-initializer-overrides -DNO_LIBCURL -DNO_ARGP "-DGLES2 1" -s MODULARIZE=1 -s EXPORT_NAME=StelWebEngine -s ALLOW_MEMORY_GROWTH=1 -s ALLOW_TABLE_GROWTH=1 --pre-js src/js/pre.js --pre-js src/js/obj.js --pre-js src/js/geojson.js --pre-js src/js/canvas.js -s RESERVED_FUNCTION_POINTERS=10 -O3 -s USE_WEBGL2=1 -s NO_EXIT_RUNTIME=1 -s "EXPORTED_FUNCTIONS=[]" -s "EXTRA_EXPORTED_RUNTIME_METHODS=['ALLOC_NORMAL','GL','UTF8ToString','_free','_malloc','addFunction','allocate','ccall','cwrap','getValue','intArrayFromString','lengthBytesUTF8','removeFunction','setValue','stringToUTF8','writeAsciiToMemory','writeArrayToMemory']" -s FILESYSTEM=0 -Isrc -Iext_src/erfa -Iext_src/json -Iext_src/uthash -Iext_src/stb -Iext_src/zlib -Iext_src/inih -Iext_src/nanovg -Iext_src/md4c -Iext_src/webp -Iext_src/webp/src ext_src/md4c/md4c.c
/workspaces/stellarium-web-engine/emsdk/upstream/emscripten/tools/scons/site_scons/site_tools/emscripten/../../../../../emcc -o build/ext_src/nanovg/nanovg.o -c -Wall -std=gnu11 -Wno-unknown-pragmas -D_GNU_SOURCE -Wno-missing-braces -DHAVE_UNISTD_H -DMD4C_USE_UTF8 -Werror -DNDEBUG -include config.h -Wno-initializer-overrides -DNO_LIBCURL -DNO_ARGP "-DGLES2 1" -s MODULARIZE=1 -s EXPORT_NAME=StelWebEngine -s ALLOW_MEMORY_GROWTH=1 -s ALLOW_TABLE_GROWTH=1 --pre-js src/js/pre.js --pre-js src/js/obj.js --pre-js src/js/geojson.js --pre-js src/js/canvas.js -s RESERVED_FUNCTION_POINTERS=10 -O3 -s USE_WEBGL2=1 -s NO_EXIT_RUNTIME=1 -s "EXPORTED_FUNCTIONS=[]" -s "EXTRA_EXPORTED_RUNTIME_METHODS=['ALLOC_NORMAL','GL','UTF8ToString','_free','_malloc','addFunction','allocate','ccall','cwrap','getValue','intArrayFromString','lengthBytesUTF8','removeFunction','setValue','stringToUTF8','writeAsciiToMemory','writeArrayToMemory']" -s FILESYSTEM=0 -Isrc -Iext_src/erfa -Iext_src/json -Iext_src/uthash -Iext_src/stb -Iext_src/zlib -Iext_src/inih -Iext_src/nanovg -Iext_src/md4c -Iext_src/webp -Iext_src/webp/src ext_src/nanovg/nanovg.c
emcc: error: linker setting ignored during compilation: 'MODULARIZE' [-Wunused-command-line-argument] [-Werror]
scons: *** [build/ext_src/erfa/erfa.o] Error 1
emcc: error: linker setting ignored during compilation: 'MODULARIZE' [-Wunused-command-line-argument] [-Werror]
scons: *** [build/ext_src/inih/ini.o] Error 1
emcc: error: linker setting ignored during compilation: 'MODULARIZE' [-Wunused-command-line-argument] [-Werror]
scons: *** [build/ext_src/json/json-builder.o] Error 1
emcc: error: linker setting ignored during compilation: 'MODULARIZE' [-Wunused-command-line-argument] [-Werror]
emcc: error: linker setting ignored during compilation: 'MODULARIZE' [-Wunused-command-line-argument] [-Werror]
emcc: error: linker setting ignored during compilation: 'MODULARIZE' [-Wunused-command-line-argument] [-Werror]
scons: *** [build/ext_src/md4c/md4c-html.o] Error 1
scons: *** [build/ext_src/json/json.o] Error 1
scons: *** [build/ext_src/md4c/entity.o] Error 1
emcc: error: linker setting ignored during compilation: 'MODULARIZE' [-Wunused-command-line-argument] [-Werror]
scons: *** [build/ext_src/md4c/md4c.o] Error 1
emcc: error: linker setting ignored during compilation: 'MODULARIZE' [-Wunused-command-line-argument] [-Werror]
scons: *** [build/ext_src/nanovg/nanovg.o] Error 1
scons: building terminated because of errors.
make: *** [Makefile:6: js] Error 2
@liudonghua123 ➜ /workspaces/stellarium-web-engine (master) $
```

</details> 

I wrote a simple bash script to test the last working emsdk, and it stopped at `2.0.18`, so the last known working version is `2.0.17`.

```bash
for i in 2.0.{11..34}; do 
    echo "testing emsdk $i"; 
    ./emsdk/emsdk install $i && source ./emsdk/emsdk_env.sh && make clean js
    if [[ $? -ne 0 ]]; then 
        echo "failed test with $i"; 
        break; 
    fi
done
```

<details><summary>Details</summary>

```
...
testing emsdk 2.0.18
Resolving SDK version '2.0.18' to 'sdk-releases-c2ac7520fad29a7937ed60ab6a95b08eb374c7ba-64bit'
Installing SDK 'sdk-releases-c2ac7520fad29a7937ed60ab6a95b08eb374c7ba-64bit'..
Skipped installing node-16.20.0-64bit, already installed.
Installing tool 'releases-c2ac7520fad29a7937ed60ab6a95b08eb374c7ba-64bit'..
Downloading: /workspaces/stellarium-web-engine/emsdk/downloads/c2ac7520fad29a7937ed60ab6a95b08eb374c7ba-wasm-binaries.tbz2 from https://storage.googleapis.com/webassembly/emscripten-releases-builds/linux/c2ac7520fad29a7937ed60ab6a95b08eb374c7ba/wasm-binaries.tbz2, 177365054 Bytes
Unpacking '/workspaces/stellarium-web-engine/emsdk/downloads/c2ac7520fad29a7937ed60ab6a95b08eb374c7ba-wasm-binaries.tbz2' to '/workspaces/stellarium-web-engine/emsdk/upstream'
Done installing tool 'releases-c2ac7520fad29a7937ed60ab6a95b08eb374c7ba-64bit'.
Done installing SDK 'sdk-releases-c2ac7520fad29a7937ed60ab6a95b08eb374c7ba-64bit'.
Setting up EMSDK environment (suppress these messages with EMSDK_QUIET=1)
Setting environment variables:
scons -c
scons: Reading SConscript files ...
scons: done reading SConscript files.
scons: Cleaning targets ...
Removed build/ext_src/erfa/erfa.o
Removed build/ext_src/inih/ini.o
Removed build/ext_src/json/json-builder.o
Removed build/ext_src/json/json.o
Removed build/ext_src/md4c/entity.o
Removed build/ext_src/md4c/md4c-html.o
Removed build/ext_src/md4c/md4c.o
Removed build/ext_src/nanovg/nanovg.o
Removed build/ext_src/webp/src/dec/alpha_dec.o
Removed build/ext_src/webp/src/dec/buffer_dec.o
Removed build/ext_src/webp/src/dec/frame_dec.o
Removed build/ext_src/webp/src/dec/idec_dec.o
Removed build/ext_src/webp/src/dec/io_dec.o
Removed build/ext_src/webp/src/dec/quant_dec.o
Removed build/ext_src/webp/src/dec/tree_dec.o
Removed build/ext_src/webp/src/dec/vp8_dec.o
Removed build/ext_src/webp/src/dec/vp8l_dec.o
Removed build/ext_src/webp/src/dec/webp_dec.o
Removed build/ext_src/webp/src/dsp/alpha_processing.o
Removed build/ext_src/webp/src/dsp/cpu.o
Removed build/ext_src/webp/src/dsp/dec.o
Removed build/ext_src/webp/src/dsp/dec_clip_tables.o
Removed build/ext_src/webp/src/dsp/filters.o
Removed build/ext_src/webp/src/dsp/lossless.o
Removed build/ext_src/webp/src/dsp/rescaler.o
Removed build/ext_src/webp/src/dsp/upsampling.o
Removed build/ext_src/webp/src/dsp/yuv.o
Removed build/ext_src/webp/src/utils/bit_reader_utils.o
Removed build/ext_src/webp/src/utils/color_cache_utils.o
Removed build/ext_src/webp/src/utils/filters_utils.o
Removed build/ext_src/webp/src/utils/huffman_utils.o
Removed build/ext_src/webp/src/utils/quant_levels_dec_utils.o
Removed build/ext_src/webp/src/utils/random_utils.o
Removed build/ext_src/webp/src/utils/rescaler_utils.o
Removed build/ext_src/webp/src/utils/thread_utils.o
Removed build/ext_src/webp/src/utils/utils.o
Removed build/ext_src/zlib/adler32.o
Removed build/ext_src/zlib/compress.o
Removed build/ext_src/zlib/crc32.o
Removed build/ext_src/zlib/deflate.o
Removed build/ext_src/zlib/gzclose.o
Removed build/ext_src/zlib/gzlib.o
Removed build/ext_src/zlib/gzread.o
Removed build/ext_src/zlib/gzwrite.o
Removed build/ext_src/zlib/infback.o
Removed build/ext_src/zlib/inffast.o
Removed build/ext_src/zlib/inflate.o
Removed build/ext_src/zlib/inftrees.o
Removed build/ext_src/zlib/trees.o
Removed build/ext_src/zlib/uncompr.o
Removed build/ext_src/zlib/zutil.o
Removed build/src/algos/bv_to_rgb.o
Removed build/src/algos/cst-boundaries.o
Removed build/src/algos/deltat.o
Removed build/src/algos/format.o
Removed build/src/algos/gust86.o
Removed build/src/algos/healpix.o
Removed build/src/algos/l1.o
Removed build/src/algos/moon.o
Removed build/src/algos/orbit.o
Removed build/src/algos/pluto.o
Removed build/src/algos/refraction.o
Removed build/src/algos/satrings.o
Removed build/src/algos/tass17.o
Removed build/src/algos/utctt.o
Removed build/src/areas.o
Removed build/src/args.o
Removed build/src/assets.o
Removed build/src/core.o
Removed build/src/designation.o
Removed build/src/eph-file.o
Removed build/src/events.o
Removed build/src/frames.o
Removed build/src/geojson_parser.o
Removed build/src/hip.o
Removed build/src/hips.o
Removed build/src/layer.o
Removed build/src/libtess2.o
Removed build/src/line_mesh.o
Removed build/src/module.o
Removed build/src/modules/atmosphere.o
Removed build/src/modules/cardinal.o
Removed build/src/modules/circle.o
Removed build/src/modules/comets.o
Removed build/src/modules/constellations.o
Removed build/src/modules/coordinates.o
Removed build/src/modules/debug.o
Removed build/src/modules/drag_selection.o
Removed build/src/modules/dso.o
Removed build/src/modules/dss.o
Removed build/src/modules/geojson.o
Removed build/src/modules/labels.o
Removed build/src/modules/landscape.o
Removed build/src/modules/lines.o
Removed build/src/modules/meteors.o
Removed build/src/modules/milkyway.o
Removed build/src/modules/minorplanets.o
Removed build/src/modules/movements.o
Removed build/src/modules/photos.o
Removed build/src/modules/planets.o
Removed build/src/modules/pointer.o
Removed build/src/modules/satellites.o
Removed build/src/modules/skycultures.o
Removed build/src/modules/stars.o
Removed build/src/mpc.o
Removed build/src/navigation.o
Removed build/src/obj.o
Removed build/src/observer.o
Removed build/src/otypes.o
Removed build/src/painter.o
Removed build/src/projection.o
Removed build/src/projections/proj_hammer.o
Removed build/src/projections/proj_mercator.o
Removed build/src/projections/proj_mollweide.o
Removed build/src/projections/proj_perspective.o
Removed build/src/projections/proj_stereographic.o
Removed build/src/render_gl.o
Removed build/src/sgp4.o
Removed build/src/shader_cache.o
Removed build/src/skybrightness.o
Removed build/src/skyculture.o
Removed build/src/sprintf.o
Removed build/src/swe.o
Removed build/src/symbols.o
Removed build/src/system.o
Removed build/src/telescope.o
Removed build/src/tests.o
Removed build/src/tonemapper.o
Removed build/src/utils/cache.o
Removed build/src/utils/fader.o
Removed build/src/utils/fps.o
Removed build/src/utils/gesture.o
Removed build/src/utils/gl.o
Removed build/src/utils/mesh.o
Removed build/src/utils/progressbar.o
Removed build/src/utils/request.o
Removed build/src/utils/request_js.o
Removed build/src/utils/swe_utils.o
Removed build/src/utils/texture.o
Removed build/src/utils/utf8.o
Removed build/src/utils/utils_json.o
Removed build/src/utils/vec.o
Removed build/src/utils/worker.o
Removed build/src/uv_map.o
Removed build/stellarium-web-engine.js
scons: done cleaning targets.
emscons scons -j8 mode=release
scons: Reading SConscript files ...
scons: done reading SConscript files.
scons: Building targets ...
/workspaces/stellarium-web-engine/emsdk/upstream/emscripten/tools/scons/site_scons/site_tools/emscripten/../../../../../emcc -o build/ext_src/erfa/erfa.o -c -Wall -std=gnu11 -Wno-unknown-pragmas -D_GNU_SOURCE -Wno-missing-braces -DHAVE_UNISTD_H -DMD4C_USE_UTF8 -Werror -DNDEBUG -include config.h -Wno-initializer-overrides -DNO_LIBCURL -DNO_ARGP "-DGLES2 1" -s MODULARIZE=1 -s EXPORT_NAME=StelWebEngine -s ALLOW_MEMORY_GROWTH=1 -s ALLOW_TABLE_GROWTH=1 --pre-js src/js/pre.js --pre-js src/js/obj.js --pre-js src/js/geojson.js --pre-js src/js/canvas.js -s RESERVED_FUNCTION_POINTERS=10 -O3 -s USE_WEBGL2=1 -s NO_EXIT_RUNTIME=1 -s "EXPORTED_FUNCTIONS=[]" -s "EXTRA_EXPORTED_RUNTIME_METHODS=['ALLOC_NORMAL','GL','UTF8ToString','_free','_malloc','addFunction','allocate','ccall','cwrap','getValue','intArrayFromString','lengthBytesUTF8','removeFunction','setValue','stringToUTF8','writeAsciiToMemory','writeArrayToMemory']" -s FILESYSTEM=0 -Isrc -Iext_src/erfa -Iext_src/json -Iext_src/uthash -Iext_src/stb -Iext_src/zlib -Iext_src/inih -Iext_src/nanovg -Iext_src/md4c -Iext_src/webp -Iext_src/webp/src ext_src/erfa/erfa.c
/workspaces/stellarium-web-engine/emsdk/upstream/emscripten/tools/scons/site_scons/site_tools/emscripten/../../../../../emcc -o build/ext_src/inih/ini.o -c -Wall -std=gnu11 -Wno-unknown-pragmas -D_GNU_SOURCE -Wno-missing-braces -DHAVE_UNISTD_H -DMD4C_USE_UTF8 -Werror -DNDEBUG -include config.h -Wno-initializer-overrides -DNO_LIBCURL -DNO_ARGP "-DGLES2 1" -s MODULARIZE=1 -s EXPORT_NAME=StelWebEngine -s ALLOW_MEMORY_GROWTH=1 -s ALLOW_TABLE_GROWTH=1 --pre-js src/js/pre.js --pre-js src/js/obj.js --pre-js src/js/geojson.js --pre-js src/js/canvas.js -s RESERVED_FUNCTION_POINTERS=10 -O3 -s USE_WEBGL2=1 -s NO_EXIT_RUNTIME=1 -s "EXPORTED_FUNCTIONS=[]" -s "EXTRA_EXPORTED_RUNTIME_METHODS=['ALLOC_NORMAL','GL','UTF8ToString','_free','_malloc','addFunction','allocate','ccall','cwrap','getValue','intArrayFromString','lengthBytesUTF8','removeFunction','setValue','stringToUTF8','writeAsciiToMemory','writeArrayToMemory']" -s FILESYSTEM=0 -Isrc -Iext_src/erfa -Iext_src/json -Iext_src/uthash -Iext_src/stb -Iext_src/zlib -Iext_src/inih -Iext_src/nanovg -Iext_src/md4c -Iext_src/webp -Iext_src/webp/src ext_src/inih/ini.c
/workspaces/stellarium-web-engine/emsdk/upstream/emscripten/tools/scons/site_scons/site_tools/emscripten/../../../../../emcc -o build/ext_src/json/json-builder.o -c -Wall -std=gnu11 -Wno-unknown-pragmas -D_GNU_SOURCE -Wno-missing-braces -DHAVE_UNISTD_H -DMD4C_USE_UTF8 -Werror -DNDEBUG -include config.h -Wno-initializer-overrides -DNO_LIBCURL -DNO_ARGP "-DGLES2 1" -s MODULARIZE=1 -s EXPORT_NAME=StelWebEngine -s ALLOW_MEMORY_GROWTH=1 -s ALLOW_TABLE_GROWTH=1 --pre-js src/js/pre.js --pre-js src/js/obj.js --pre-js src/js/geojson.js --pre-js src/js/canvas.js -s RESERVED_FUNCTION_POINTERS=10 -O3 -s USE_WEBGL2=1 -s NO_EXIT_RUNTIME=1 -s "EXPORTED_FUNCTIONS=[]" -s "EXTRA_EXPORTED_RUNTIME_METHODS=['ALLOC_NORMAL','GL','UTF8ToString','_free','_malloc','addFunction','allocate','ccall','cwrap','getValue','intArrayFromString','lengthBytesUTF8','removeFunction','setValue','stringToUTF8','writeAsciiToMemory','writeArrayToMemory']" -s FILESYSTEM=0 -Isrc -Iext_src/erfa -Iext_src/json -Iext_src/uthash -Iext_src/stb -Iext_src/zlib -Iext_src/inih -Iext_src/nanovg -Iext_src/md4c -Iext_src/webp -Iext_src/webp/src ext_src/json/json-builder.c
/workspaces/stellarium-web-engine/emsdk/upstream/emscripten/tools/scons/site_scons/site_tools/emscripten/../../../../../emcc -o build/ext_src/json/json.o -c -Wall -std=gnu11 -Wno-unknown-pragmas -D_GNU_SOURCE -Wno-missing-braces -DHAVE_UNISTD_H -DMD4C_USE_UTF8 -Werror -DNDEBUG -include config.h -Wno-initializer-overrides -DNO_LIBCURL -DNO_ARGP "-DGLES2 1" -s MODULARIZE=1 -s EXPORT_NAME=StelWebEngine -s ALLOW_MEMORY_GROWTH=1 -s ALLOW_TABLE_GROWTH=1 --pre-js src/js/pre.js --pre-js src/js/obj.js --pre-js src/js/geojson.js --pre-js src/js/canvas.js -s RESERVED_FUNCTION_POINTERS=10 -O3 -s USE_WEBGL2=1 -s NO_EXIT_RUNTIME=1 -s "EXPORTED_FUNCTIONS=[]" -s "EXTRA_EXPORTED_RUNTIME_METHODS=['ALLOC_NORMAL','GL','UTF8ToString','_free','_malloc','addFunction','allocate','ccall','cwrap','getValue','intArrayFromString','lengthBytesUTF8','removeFunction','setValue','stringToUTF8','writeAsciiToMemory','writeArrayToMemory']" -s FILESYSTEM=0 -Isrc -Iext_src/erfa -Iext_src/json -Iext_src/uthash -Iext_src/stb -Iext_src/zlib -Iext_src/inih -Iext_src/nanovg -Iext_src/md4c -Iext_src/webp -Iext_src/webp/src ext_src/json/json.c
/workspaces/stellarium-web-engine/emsdk/upstream/emscripten/tools/scons/site_scons/site_tools/emscripten/../../../../../emcc -o build/ext_src/md4c/entity.o -c -Wall -std=gnu11 -Wno-unknown-pragmas -D_GNU_SOURCE -Wno-missing-braces -DHAVE_UNISTD_H -DMD4C_USE_UTF8 -Werror -DNDEBUG -include config.h -Wno-initializer-overrides -DNO_LIBCURL -DNO_ARGP "-DGLES2 1" -s MODULARIZE=1 -s EXPORT_NAME=StelWebEngine -s ALLOW_MEMORY_GROWTH=1 -s ALLOW_TABLE_GROWTH=1 --pre-js src/js/pre.js --pre-js src/js/obj.js --pre-js src/js/geojson.js --pre-js src/js/canvas.js -s RESERVED_FUNCTION_POINTERS=10 -O3 -s USE_WEBGL2=1 -s NO_EXIT_RUNTIME=1 -s "EXPORTED_FUNCTIONS=[]" -s "EXTRA_EXPORTED_RUNTIME_METHODS=['ALLOC_NORMAL','GL','UTF8ToString','_free','_malloc','addFunction','allocate','ccall','cwrap','getValue','intArrayFromString','lengthBytesUTF8','removeFunction','setValue','stringToUTF8','writeAsciiToMemory','writeArrayToMemory']" -s FILESYSTEM=0 -Isrc -Iext_src/erfa -Iext_src/json -Iext_src/uthash -Iext_src/stb -Iext_src/zlib -Iext_src/inih -Iext_src/nanovg -Iext_src/md4c -Iext_src/webp -Iext_src/webp/src ext_src/md4c/entity.c
/workspaces/stellarium-web-engine/emsdk/upstream/emscripten/tools/scons/site_scons/site_tools/emscripten/../../../../../emcc -o build/ext_src/md4c/md4c-html.o -c -Wall -std=gnu11 -Wno-unknown-pragmas -D_GNU_SOURCE -Wno-missing-braces -DHAVE_UNISTD_H -DMD4C_USE_UTF8 -Werror -DNDEBUG -include config.h -Wno-initializer-overrides -DNO_LIBCURL -DNO_ARGP "-DGLES2 1" -s MODULARIZE=1 -s EXPORT_NAME=StelWebEngine -s ALLOW_MEMORY_GROWTH=1 -s ALLOW_TABLE_GROWTH=1 --pre-js src/js/pre.js --pre-js src/js/obj.js --pre-js src/js/geojson.js --pre-js src/js/canvas.js -s RESERVED_FUNCTION_POINTERS=10 -O3 -s USE_WEBGL2=1 -s NO_EXIT_RUNTIME=1 -s "EXPORTED_FUNCTIONS=[]" -s "EXTRA_EXPORTED_RUNTIME_METHODS=['ALLOC_NORMAL','GL','UTF8ToString','_free','_malloc','addFunction','allocate','ccall','cwrap','getValue','intArrayFromString','lengthBytesUTF8','removeFunction','setValue','stringToUTF8','writeAsciiToMemory','writeArrayToMemory']" -s FILESYSTEM=0 -Isrc -Iext_src/erfa -Iext_src/json -Iext_src/uthash -Iext_src/stb -Iext_src/zlib -Iext_src/inih -Iext_src/nanovg -Iext_src/md4c -Iext_src/webp -Iext_src/webp/src ext_src/md4c/md4c-html.c
/workspaces/stellarium-web-engine/emsdk/upstream/emscripten/tools/scons/site_scons/site_tools/emscripten/../../../../../emcc -o build/ext_src/md4c/md4c.o -c -Wall -std=gnu11 -Wno-unknown-pragmas -D_GNU_SOURCE -Wno-missing-braces -DHAVE_UNISTD_H -DMD4C_USE_UTF8 -Werror -DNDEBUG -include config.h -Wno-initializer-overrides -DNO_LIBCURL -DNO_ARGP "-DGLES2 1" -s MODULARIZE=1 -s EXPORT_NAME=StelWebEngine -s ALLOW_MEMORY_GROWTH=1 -s ALLOW_TABLE_GROWTH=1 --pre-js src/js/pre.js --pre-js src/js/obj.js --pre-js src/js/geojson.js --pre-js src/js/canvas.js -s RESERVED_FUNCTION_POINTERS=10 -O3 -s USE_WEBGL2=1 -s NO_EXIT_RUNTIME=1 -s "EXPORTED_FUNCTIONS=[]" -s "EXTRA_EXPORTED_RUNTIME_METHODS=['ALLOC_NORMAL','GL','UTF8ToString','_free','_malloc','addFunction','allocate','ccall','cwrap','getValue','intArrayFromString','lengthBytesUTF8','removeFunction','setValue','stringToUTF8','writeAsciiToMemory','writeArrayToMemory']" -s FILESYSTEM=0 -Isrc -Iext_src/erfa -Iext_src/json -Iext_src/uthash -Iext_src/stb -Iext_src/zlib -Iext_src/inih -Iext_src/nanovg -Iext_src/md4c -Iext_src/webp -Iext_src/webp/src ext_src/md4c/md4c.c
/workspaces/stellarium-web-engine/emsdk/upstream/emscripten/tools/scons/site_scons/site_tools/emscripten/../../../../../emcc -o build/ext_src/nanovg/nanovg.o -c -Wall -std=gnu11 -Wno-unknown-pragmas -D_GNU_SOURCE -Wno-missing-braces -DHAVE_UNISTD_H -DMD4C_USE_UTF8 -Werror -DNDEBUG -include config.h -Wno-initializer-overrides -DNO_LIBCURL -DNO_ARGP "-DGLES2 1" -s MODULARIZE=1 -s EXPORT_NAME=StelWebEngine -s ALLOW_MEMORY_GROWTH=1 -s ALLOW_TABLE_GROWTH=1 --pre-js src/js/pre.js --pre-js src/js/obj.js --pre-js src/js/geojson.js --pre-js src/js/canvas.js -s RESERVED_FUNCTION_POINTERS=10 -O3 -s USE_WEBGL2=1 -s NO_EXIT_RUNTIME=1 -s "EXPORTED_FUNCTIONS=[]" -s "EXTRA_EXPORTED_RUNTIME_METHODS=['ALLOC_NORMAL','GL','UTF8ToString','_free','_malloc','addFunction','allocate','ccall','cwrap','getValue','intArrayFromString','lengthBytesUTF8','removeFunction','setValue','stringToUTF8','writeAsciiToMemory','writeArrayToMemory']" -s FILESYSTEM=0 -Isrc -Iext_src/erfa -Iext_src/json -Iext_src/uthash -Iext_src/stb -Iext_src/zlib -Iext_src/inih -Iext_src/nanovg -Iext_src/md4c -Iext_src/webp -Iext_src/webp/src ext_src/nanovg/nanovg.c
shared:INFO: (Emscripten: Running sanity checks)
emcc: error: EXTRA_EXPORTED_RUNTIME_METHODS is deprecated, please use EXPORTED_RUNTIME_METHODS instead [-Wdeprecated] [-Werror]
emcc: error: EXTRA_EXPORTED_RUNTIME_METHODS is deprecated, please use EXPORTED_RUNTIME_METHODS instead [-Wdeprecated] [-Werror]
emcc: error: EXTRA_EXPORTED_RUNTIME_METHODS is deprecated, please use EXPORTED_RUNTIME_METHODS instead [-Wdeprecated] [-Werror]
emcc: error: EXTRA_EXPORTED_RUNTIME_METHODS is deprecated, please use EXPORTED_RUNTIME_METHODS instead [-Wdeprecated] [-Werror]
emcc: error: EXTRA_EXPORTED_RUNTIME_METHODS is deprecated, please use EXPORTED_RUNTIME_METHODS instead [-Wdeprecated] [-Werror]
emcc: error: EXTRA_EXPORTED_RUNTIME_METHODS is deprecated, please use EXPORTED_RUNTIME_METHODS instead [-Wdeprecated] [-Werror]
emcc: error: EXTRA_EXPORTED_RUNTIME_METHODS is deprecated, please use EXPORTED_RUNTIME_METHODS instead [-Wdeprecated] [-Werror]
emcc: error: EXTRA_EXPORTED_RUNTIME_METHODS is deprecated, please use EXPORTED_RUNTIME_METHODS instead [-Wdeprecated] [-Werror]
scons: *** [build/ext_src/md4c/md4c-html.o] Error 1
scons: *** [build/ext_src/erfa/erfa.o] Error 1
scons: *** [build/ext_src/md4c/md4c.o] Error 1
scons: *** [build/ext_src/json/json-builder.o] Error 1
scons: *** [build/ext_src/md4c/entity.o] Error 1
scons: *** [build/ext_src/nanovg/nanovg.o] Error 1
scons: *** [build/ext_src/json/json.o] Error 1
scons: *** [build/ext_src/inih/ini.o] Error 1
scons: building terminated because of errors.
make: *** [Makefile:6: js] Error 2
failed test with 2.0.18
```

</details>